x/stake/types/delegation.go

```diff
@@ -17,7 +18,41 @@ type Delegation struct {
 	Height        int64       `json:"height"` // Last height bond updated
 }
 
-// Equal returns a boolean determining if two Delegation types are identical.
+type delegationValue struct {
+	Shares sdk.Rat
+	Height int64
+}
+
+// return the delegation without fields contained within the key for the store
+func MarshalDelegation(cdc *wire.Codec, delegation Delegation) []byte {
+	val := delegationValue{
+		delegation.Shares,
+		delegation.Height,
+	}
+	return cdc.MustMarshalBinary(val)
+}
+
+// return the delegation without fields contained within the key for the store
+func UnmarshalDelegation(cdc *wire.Codec, key, value []byte) Delegation {
+	var storeValue delegationValue
+	cdc.MustUnmarshalBinary(value, &storeValue)
+
+	addrs := key[1:] // remove prefix bytes
+	if len(addrs) != 40 {
+		panic("unexpected key length")
+	}
+	delAddr := sdk.Address(addrs[:20])
```

Comment:
Can address length be a constant declared somewhere?

Reply:
yup - should be close to `sdk.Address`

Reply:
addressed